source/rfc/ansible-style-guide.rst

```diff
@@ -0,0 +1,183 @@
+Ansible Style Guide
+===================
+
+Purpose
+-------
+
+The purpose of this document is to define and describe the style guide to be used for Ansible
+in contrail-project-config and contrail-zuul-jobs projects. Topics include but are not limited
+to: role tasks separation between task files, task definition, interpolation syntax etc.
+
+This guide is inspired by `whitecloud's ansible style guide <https://github.com/whitecloud/ansible-styleguide>`_.
+
+Rules
+-----
+
+YAML File Format
+^^^^^^^^^^^^^^^^
+
+YAML files should begin with the YAML marker `---` marking the start of the document and end with a newline.
+
+.. code-block:: yaml
+
+  ---
+
+  - name: restart apache
+    service:
+      name: 'apache2'
+      state: 'restarted'
+
+Hosts Declaration
+^^^^^^^^^^^^^^^^^
+
+Use the following order for hosts declaration:
+  * `hosts:` declaration
+  * host options in alphabetical order
+  * pre_tasks
+  * roles
+  * tasks
+
+.. code-block:: yaml
+
+  - hosts: webservers
+    remote_user: 'centos'
+    vars:
+      tomcat_state: 'started'
+    pre_tasks:
+      - name: set the timezone to America/Boise
+        lineinfile:
+          dest: '/etc/environment'
+          line: 'TZ=America/Boise'
+          state: 'present'
+        become: true
+    roles:
+      - { role: 'tomcat', tags: 'tomcat' }
+    tasks:
+      - name: start the tomcat service
+        service:
+          name: 'tomcat'
+        state: '{{ tomcat_state }}'
+
+Task Declaration
+^^^^^^^^^^^^^^^^
+
+* Lowercase the first letter of a task name.
+* Use the map syntax. Use only one space after the colon.
+* Use the following order for task declaration:
+  * task name
+  * task map declaration
+  * loop operators
+  * task options in alphabetical order
+
+.. code-block:: yaml
+
+  - name: template sensu configuration
+    template:
+      src: '{{ item }}.conf.j2'
+      dest: '/etc/sensu/{{ item }}.conf'
+      mode: '0644'
+      owner: 'sensu'
+      group: 'sensu'
+    with_items:
+      - sensu-api
+      - sensu-client
+      - sensu-server
+    become: true
+
+Role File Structure
+^^^^^^^^^^^^^^^^^^^
+
+The main.yaml file in `tasks` directory should contain only `include_tasks:` statements, in alphabetical
```

Comment:
I think there were objections to that

Reply:
Can't remember any specifics, waiting for other reviews.

Reply:
I think that this will add unnecessary level of depth, most of the time we don't need multiple task files inside a single role.

Reply:
Rephrased the paragraph. @wurbanski @jluk-codi let me know whether it's good now or should I remove it entirely.